napari/_tests/utils.py

```diff
@@ -120,9 +121,30 @@ def add_layer_by_type(viewer, layer_type, data, visible=True):
     return layer2addmethod[layer_type](viewer, data, visible=visible)
 
 
+def are_objects_equal(object1, object2):
+    """
+    compare two (collections of) arrays or other objects for equality. Ignores nan.
+    """
+    if isinstance(object1, (list, tuple)):
+        items = zip(object1, object2)
+    elif isinstance(object1, dict):
+        items = [(value, object2[key]) for key, value in object1.items()]
+    else:
+        items = [(object1, object2)]
+
+    try:
+        return np.all(
+            [np.array_equal(a1, a2, equal_nan=True) for a1, a2 in items]
+        )
+    except TypeError:
+        # np.array_equal fails for arrays of type `object` (e.g: strings)
+        return np.all([a1 == a2 for a1, a2 in items])
```

Comment:
Here we compare the equality of two objects that may be collections of things or single things, also accounting for the presence of `nan`. I thought tests looked more intuitive with this logic separated out, but I'm happy to split up a bit if you think this should not live all in the same function.